nixio/source.py

```diff
@@ -48,6 +49,51 @@ def create_source(self, name, type_):
         src = Source.create_new(self.file, self, sources, name, type_)
         return src
 
+    @property
+    def parent_block(self):
+        """
+        Returns the block this source is contained in.
+
+        :returns: the Block containing this source
+        :rtype: Block
+        """
+        if self._parent_block is not None:
+            return self._parent_block
+        maybe_block = self._parent
+        if not hasattr(maybe_block, "data_arrays"):
+            maybe_block = maybe_block.parent_block
+        self._parent_block = maybe_block
+        return self._parent_block
+
+    @staticmethod
+    def _find_parent_recursive(source, name_or_id_of_child):
+        if name_or_id_of_child in source.sources:
+            return source
+        else:
+            for s in source.sources:
+                s = Source._find_parent_recursive(s, name_or_id_of_child)
+                if s is not None:
+                    return s
+        return None
```

Comment:
Could this be an instance method with the first arg as `self` instead of a `source`?  Or is the source not always an instantiated NIX object and could just be an H5 reference?

Also, it might be safer to only use IDs for searching and not names.  It's possible for multiple sources to have the same name if they're not on the same level and this could find the wrong parent.
I see the actual implementation below does use the ID, so maybe the argument/variable name could reflect this to make it clear and maybe check if it's a UUID.

Reply:
@achilleas-k you are right. Please check the current version and regardless of what GitHub claims, of course I did not dismiss your review ;)

Reply:
haha
yeah I think it dismisses automatically when the code changes.